ginivision/src/main/java/net/gini/android/vision/noresults/NoResultsFragmentImpl.java

```diff
@@ -1,50 +1,72 @@
 package net.gini.android.vision.noresults;
 
 
+import static android.view.View.GONE;
+
 import static net.gini.android.vision.internal.util.ActivityHelper.forcePortraitOrientationOnPhones;
 
-import android.content.Context;
+import android.content.Intent;
 import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
+import net.gini.android.vision.Document;
 import net.gini.android.vision.R;
 import net.gini.android.vision.internal.ui.FragmentImplCallback;
 
 class NoResultsFragmentImpl {
 
+    private static final NoResultsFragmentListener NO_OP_LISTENER = new NoResultsFragmentListener() {
+        @Override
+        public void onBackToCameraPressed() {
+        }
+    };
+
     private final FragmentImplCallback mFragment;
+    private final Document mDocument;
     private NoResultsFragmentListener mListener;
 
-    NoResultsFragmentImpl(final FragmentImplCallback fragment) {
+    NoResultsFragmentImpl(@NonNull final FragmentImplCallback fragment,
+            @NonNull final Document document) {
         mFragment = fragment;
+        mDocument = document;
     }
 
-    public void onCreate(final Bundle savedInstanceState) {
-        forcePortraitOrientationOnPhones(mFragment.getActivity());
+    void setListener(@Nullable NoResultsFragmentListener listener) {
+        if (listener == null) {
+            mListener = NO_OP_LISTENER;
+        } else {
+            mListener = listener;
+        }
     }
 
-    void onAttach(Context context) {
-        try {
-            mListener = (NoResultsFragmentListener) context;
-        } catch (ClassCastException e) {
-            throw new ClassCastException(
-                    context.toString() + " must implement NoResultsFragmentListener");
-        }
+    void onCreate(final Bundle savedInstanceState) {
+        forcePortraitOrientationOnPhones(mFragment.getActivity());
     }
 
     View onCreateView(final LayoutInflater inflater, final ViewGroup container,
             final Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.gv_fragment_noresults, container, false);
         View backButton = view.findViewById(R.id.gv_button_no_results_back);
-        backButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(final View view) {
-                mListener.onBackToCameraPressed();
-            }
-        });
+        if (isDocumentFromCameraScreen()) {
+            backButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(final View view) {
+                    mListener.onBackToCameraPressed();
+                }
+            });
+        } else {
+            backButton.setVisibility(GONE);
```

Comment:
Maybe I have misunderstood something about we discussed before, but the back button should be replaced with a close button, right?

Reply:
🤔 really? I understood we should simply hide the button. We also don't have a close button design.

Reply:
But how do they exit the screen then? Because if they come from Open with there is no button on the bottom.
I'm using the same "X" that we use to exit the ScreenAPI. Maybe on Android is not needed since there is a physical button to go back.